keystorage/yubihsm2/src/test/java/tech/pegasys/signers/yubihsm2/YubiHsm2Test.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.signers.yubihsm2;
+
+import java.util.Optional;
+
+import org.apache.commons.lang3.time.StopWatch;
+import org.assertj.core.api.Assertions;
+import org.assertj.core.api.Assumptions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+// manual dev test -
+// Make sure yubihsm2 is plugged in
+// for http/https url scheme, make sure connector daemon is running
+// for yhusb://serial=123456 url scheme, no connector daemon is required.
+// uncomment system properties in setup method and modify accordingly
+class YubiHsm2Test {
+  private YubiHsm2 yubiHsm2;
+
+  @BeforeEach
+  void setup() {
+    // System.setProperty("yubihsmConnector", "yhusb://serial=123456");
+    // System.setProperty("yubiHsmAuthId", "2");
+    // System.setProperty("yubiHsmPassword", "password");
+    // System.setProperty("yubiHsmPath", "/Users/dev/yubihsm2-sdk/bin");
+
+    final String yubihsmConnector = System.getProperty("yubihsmConnector");
+    final String yubiHsmAuthId = System.getProperty("yubiHsmAuthId");
+    final String yubiHsmPassword = System.getProperty("yubiHsmPassword");
+    final String yubiShellPath = System.getProperty("yubiHsmPath");
+    final Optional<String> caCert = Optional.empty();
+    final Optional<String> proxy = Optional.empty();
+
+    Assumptions.assumeThat(yubihsmConnector).isNotEmpty();
+    Assumptions.assumeThat(yubiHsmAuthId).isNotEmpty();
+    Assumptions.assumeThat(yubiHsmPassword).isNotEmpty();
+    Assumptions.assumeThat(yubiShellPath).isNotEmpty();
+
+    yubiHsm2 =
+        new YubiHsm2(
+            yubihsmConnector,
+            Short.parseShort(yubiHsmAuthId),
+            yubiHsmPassword,
+            Optional.of(yubiShellPath),
+            caCert,
+            proxy);
+  }
+
+  @Test
+  void validDataIsReturnedFromYubiHSM() {
+    final String[] expectedKeys =
+        new String[] {
+          "000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f",
```

Comment:
are these keys are specific to your test device? is there a setup script to get these into the yubihsm?